chia/util/write_file.py

```diff
@@ -0,0 +1,24 @@
+import os
+import shutil
+
+from pathlib import Path
+from typing import Union
+
+
+def write_file(file_path: Path, data: Union[str, bytes], *, file_mode: int = 0o600):
+    """
+    Writes the provided data to a temporary file and then moves it to the final destination.
+    """
+
+    dir_perms: int = 0o700
+    # Create the parent directory if necessary
+    os.makedirs(os.path.dirname(file_path), mode=dir_perms, exist_ok=True)
+
+    temp_path: Path = file_path.with_suffix("." + str(os.getpid()))
+    mode: str = "w" if type(data) == str else "wb"
```

Comment:
I would have expected you would need `w+` and `wb+` respectively. what if a previous error left the file (with the pid suffix) around? it would need to be truncated.

There should probably be a test for that.

Reply:
I've switched it to use w+ and w+b. Existing file issue has been worked-around by using the tempfile functionality to guarantee a uniquely named temp file.